src/lib/notifications/backgroundFetch.native.js

```diff
@@ -32,59 +32,71 @@ export const initBGFetch = once((goodWallet, userStorage) => {
     log.info('[BackgroundFetch] taskId: ', taskId)
 
     const isLoggedIn = await AsyncStorage.getItem(IS_LOGGED_IN)
+    const oldNotifications = await AsyncStorage.getItem(OLD_NOTIFICATIONS)
 
     log.info('isLoggedIn', isLoggedIn)
 
     if (!isLoggedIn) {
       return BackgroundFetch.finish(taskId)
     }
 
-    try {
-      await hasConnection()
-      await onFeedReady(userStorage)
-    } catch (e) {
-      return BackgroundFetch.finish(taskId)
-    }
-
-    const lastFeedCheck = userStorage.userProperties.get('lastSeenFeedNotification')
-    const feed = await userStorage.getFeedPage(20, true)
-
-    log.info('lastFeedCheck', lastFeedCheck)
-    log.info('feed', feed)
-
-    const hasNewPayment = (type, status) => {
-      return type === 'receive' && status === 'completed'
-    }
-
-    const hasNewPaymentWithdraw = (type, status) => {
-      return type === 'send' && status === 'completed'
+    const checkNotifications = async isInitialCall => {
+      try {
+        await hasConnection()
+        await onFeedReady(userStorage)
+      } catch (e) {
+        return BackgroundFetch.finish(taskId)
+      }
+
+      const lastFeedCheck = userStorage.userProperties.get('lastSeenFeedNotification')
+      const feed = await userStorage.getFeedPage(20, true)
+
+      log.info('lastFeedCheck', lastFeedCheck)
+      log.info('feed', feed)
+
+      const hasNewPayment = (type, status) => {
+        return type === 'receive' && status === 'completed'
+      }
+
+      const hasNewPaymentWithdraw = (type, status) => {
+        return type === 'send' && status === 'completed'
+      }
+
+      const newFeeds = feed.filter(feedItem => {
+        const { date, type, status } = feedItem
+        const feedDate = moment(new Date(date)).valueOf()
+        const isActual =
+          (hasNewPayment(type, status) || hasNewPaymentWithdraw(type, status)) &&
+          lastFeedCheck < feedDate &&
+          !includes(oldNotifications, feedItem.id)
+        return isActual && feedItem
+      })
+
+      userStorage.userProperties.safeSet('lastSeenFeedNotification', Date.now())
+
+      log.info('new feed items', { newFeeds })
+
+      log.info('pushing local notifications for feed items:', { total: newFeeds.length })
+
+      newFeeds.map(async feed => {
+        if (!isInitialCall) {
+          PushNotification.localNotification({
```

Comment:
looks like demo implementation, not satisfies the definition of done. 
i'm converting it to the draft

please check the complete list of the events we need to notify: 
https://github.com/GoodDollar/GoodDAPP/issues/3768

once they will be implemented mark as ready again